code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -51,6 +55,88 @@
 	add_fingerprint(user)
 	return
 
+/obj/item/weapon/melee/energy/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
+	if(active && default_parry_check(user, attacker, damage_source) && prob(50))
+		user.visible_message("<span class='danger'>\The [user] parries [attack_text] with \the [src]!</span>")
+
+//		Disabled because lag. Immense amounts of lag.
+//		var/datum/effect/effect/system/spark_spread/spark_system = new /datum/effect/effect/system/spark_spread()
+//		spark_system.set_up(5, 0, user.loc)
+//		spark_system.start()
+		playsound(user.loc, 'sound/weapons/blade1.ogg', 50, 1)
+		return 1
+	else
+
+		if(!active)
+			return 0 //turn it on first!
+
+		if(user.incapacitated())
+			return 0
+
+		//block as long as they are not directly behind us
+		var/bad_arc = reverse_direction(user.dir) //arc of directions from which we cannot block
+		if(check_shield_arc(user, bad_arc, damage_source, attacker))
+
+			if(prob(base_block_chance))
+				var/datum/effect/effect/system/spark_spread/spark_system = getFromPool(/datum/effect/effect/system/spark_spread)
+				spark_system.set_up(5, 0, user.loc)

```

Comment:
For reference: https://github.com/Aurorastation/Aurora.3/blob/development/code/modules/effects/sparks/_spark_docs.dm